### DISCUSSION TITLE
CPMG fitting

### DISCUSSION THREAD

**Discussion by xveesi** [gbouvignies/ChemEx#295] [Q&A] [answered]
Hi,
I am trying to fit a H2C 13CH3 CPMG data. Th fitting is not smooth. Please find the comparison between PINT and chemex. How can I rectify? 
Thanks. 
[I225.pdf](https://github.com/user-attachments/files/17832485/I225.pdf)

**Comment by gbouvignies** [comment, marked as answer]
Hi xveesi,

Thanks for reaching out and for sharing your data. The wavy nature of the ChemEx profile is not necessarily a cause for concern. In fact, it is often expected in many cases and reflects the accurate physics modeled by ChemEx, particularly in the context of off-resonance effects and pulse imperfections.

ChemEx uses a detailed Bloch-McConnell approach, which integrates the effects of all pulses and accounts for the different relaxation pathways during the pulse sequence. These effects can lead to subtle variations in effective relaxation rates as a function of the CPMG frequency, manifesting as the observed waviness.

This behavior is intrinsic to ChemEx’s modeling and doesn’t imply poor fit or inaccurate results. Instead, it often provides a more realistic depiction of the experimental system than the smoother Carver-Richards curves that might overlook details.

If you'd like, I can help investigate further to ensure everything is set up optimally for your analysis. To do so, it would be helpful to look at your experiment.toml, fitted.toml, fixed.toml, and constrained.toml files to understand the parameter settings and constraints applied in the fitting process.

Let me know if you'd like to proceed or have additional questions!

Best,
Guillaume

**Comment by xveesi** [comment]
Hi Guillaume, 
Thanks for your detailed response. Could you please explain how the program calculates for pulse imperfections. In the experiment section I give only the pw90 length and so somewhere it should back calculate to see my pulse length is not ideal. Just trying to understand. Because, the power level for the pulse is not given. Then how does it know that my pulse is imperfect! 

Please find the method.toml,  experiment.toml, fitted.toml, fixed.toml, and constrained.toml  attached here. 

Thanks again. 

Best,
Siva. 
[data_files.zip](https://github.com/user-attachments/files/17844049/data_files.zip)

**Reply by gbouvignies** [reply]
Hi Siva,

Thank you for sharing your files and for your follow-up question.

When I mention "pulse imperfections", I’m referring to the fact that ChemEx does not assume pulses to be perfectly on-resonance or infinitely short. The pulse power is inferred from the provided 90° pulse width (pw90) and assumes that the pulses are properly calibrated for the given experiment.

For a detailed explanation of the formalism used in ChemEx to model pulse sequences, I highly recommend the following paper:

Helgstrand, M., Härd, T. & Allard, P. *Simulations of NMR pulse sequences during equilibrium and non-equilibrium chemical exchange.* **J Biomol NMR** 18, 49–63 (2000). https://doi.org/10.1023/A:1008309220156

This paper outlines the mathematical framework, including how off-resonance effects and finite pulse durations are incorporated.

Looking at your parameter files, I noticed something that may explain some of the issues you’ve observed. Specifically, the chemical shift of state A (CS_A) is set to 0 ppm for all your resonances, while the carrier frequency is specified as 10 ppm in your experiment.toml file. This results in all profiles being calculated with a 10 ppm offset from the carrier frequency, which may not reflect the actual experimental conditions.

To ensure accurate modeling, please verify and adjust the following:
  1. Carrier Frequency: Confirm that the “carrier” parameter in your experiment.toml file corresponds to the actual carrier frequency used during your experiments (the center of the spectrum used to make the peak list).
  2. Chemical Shifts: Ensure that the CS_A parameter in the relevant .toml file reflects the correct chemical shifts for each resonance.

Making these adjustments should provide a more accurate representation of your experimental system. Please let me know if you encounter further issues or have additional questions.

**Comment by xveesi** [comment]
Hi Guillaume, 

Thanks for your prompt response. I got the gist of how the program includes pulse imperfections and offset parameters. Coming back to the issue of my fitting, I do give the CS_A in the original parameter files. Somehow it returns to 0 ppm for all after the calculations. It may be something silly mistake I am making.  I can't figure it out at the moment. I am attaching the entire fitting for 4 residues. Please have a look and let me know. 

Thanks in advance. 

Kind regards,
Siva. 
[cpmg_CH3_13C_H2C.zip](https://github.com/user-attachments/files/17899840/cpmg_CH3_13C_H2C.zip)

**Reply by gbouvignies** [reply]
Hi Siva,

To set the CS_A of the carbons correctly, you need to name the associated spins accordingly: for example, “I225” should be changed to “I225C.” I’ve updated this in the parameters.toml file and also made the change in the 700mhz.toml and 900mhz.toml files. However, note that it isn’t strictly necessary for these two files, as ChemEx automatically adjusts the profile names in these settings.

I’ve attached the updated results. The profiles are much smoother now that the resonance frequencies are being properly accounted for.

Let me know if you encounter any other issues!

**Attachment:** [cpmg_CH3_13C_H2C.zip](https://github.com/user-attachments/files/17901285/cpmg_CH3_13C_H2C.zip)

**Comment by xveesi** [comment]
Thanks a lot.  I never would have thought that naming would be the error!
